File: savevia-optimizer/src/main/java/com/savevia/optimizer/entity/MerchantCategory.java
package com.savevia.optimizer.entity;

import lombok.Data;
import java.time.LocalDateTime;

@Data
public class MerchantCategory {
    private Long id;
    private String merchantPattern;
    private String category;
    private Integer priority;
    private Boolean isActive;
    private LocalDateTime createdAt;
}

File: savevia-ai/app/models/merchant_category.py
from datetime import datetime

from sqlalchemy import BigInteger, Boolean, DateTime, Integer, String, func
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base


class MerchantCategory(Base):
    """MCC / merchant-name → category mapping rules."""

    __tablename__ = "merchant_categories"

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True, autoincrement=True)
    merchant_pattern: Mapped[str] = mapped_column(String(200), nullable=False)
    category: Mapped[str] = mapped_column(String(50), nullable=False, index=True)

    priority: Mapped[int | None] = mapped_column(Integer, default=0, index=True)
    is_active: Mapped[bool | None] = mapped_column(Boolean, default=True)
    created_at: Mapped[datetime | None] = mapped_column(
        DateTime, server_default=func.current_timestamp()
    )
